cloud_functions/fastly_announce_purge/src/main.py

```diff
@@ -42,3 +53,97 @@ def purge_for_announce(cloud_event: CloudEvent):
         else:
             logger.warning(f"Announcement event caught, but no environment to purge cache. ENVIRONMENT: {environment}")
 
+
+def _purge_announced_papers():
+    """this function purges fastly's data for papers that have been created or updated since the last announcement"""
+    announcements= _get_days_announcements()
+    keys=_process_announcements(announcements)
+
+    #send purge request(s) to appropriate fastly services
+    environment = os.environ.get('ENVIRONMENT')
+    if environment == "PRODUCTION":
+        purge_fastly_keys(keys)
+        purge_fastly_keys(keys,"export.arxiv.org")
+    elif environment == "DEVELOPMENT":
+        purge_fastly_keys(keys, "browse.dev.arxiv.org")
+    elif environment == "TESTING":
+        logger.info(f"In TESTING enviroment. Would have purged keys {len(keys)}: {keys}\nAbove keys ({len(keys)}) not purged, in TESTING enviroment.")
+    return
+
+def _get_days_announcements()-> List[Tuple[str, int, str, str, str]]:
```

Comment:
For future consideration: maybe we should have a fn like this in base?

Reply:
I think that makes sense. The base should always know where the announcement days are (perhaps even informed by a holidays table, which we still need a ticket for prb.)

Reply:
for clarity this doesn't know when announcement days are ahead of time, it looks them up in the mailing table after the fact. I'm not sure what putting a function into base would look like because different systems need different pieces of info.